include/circt/Dialect/FIRRTL/FIRRTLTypesImpl.td

```diff
@@ -417,6 +417,19 @@ def RefImpl : FIRRTLImplType<"Ref",
   }];
 }
 
+def LHSTypeImpl : FIRRTLImplType<"LHS", [], "::circt::firrtl::FIRRTLBaseType"> {
+  let summary = "A wrapper for LHS types.";
+  let description = [{
+    A LHS type is a type usable for the destination of a strict connect
+    and for field indexing.  No other operations are valid.
+  }];
+  let parameters = (ins "FIRRTLBaseType":$type);
```

Comment:
Should this be constrained to be passive, no analog?

Judging by direction re:strictconnect -- probably also only usable for known-width and so on?

EDIT: maybe that's what the modified SizedPassiveType is for?

LHS-of-const seems not-the-most-useful but maybe not inherently invalid (should be able to at least have LHS of a bundle that has const in it for writing to the non-const portions...?), WDYT?

Reply:
LHS-of-const seems a natural fallout of using connect to assign values to const.  Which isn't any different from now.

Reply:
I don't think known-width adds anything.

Reply:
Oh, not sure what I was thinking.  So the source/destination are same type /now/ but may not be in the future -- so we'll insert the pads or whatever is needed when, e.g., doing width inference.  I think that's what we do already for matchingconnect?

Reply:
yes, matching/strict connect have type identical constraints, including pre-inference.  a z/sext : known -> unknown captures the extension semantics, not connect.